types/options.go

```diff
@@ -457,8 +457,11 @@ func ReloadConfigurationFile(configFile string, storeOptions *StoreOptions) erro
 	if config.Storage.RootlessStoragePath != "" {
 		storeOptions.RootlessStoragePath = config.Storage.RootlessStoragePath
 	}
-	for _, s := range config.Storage.Options.AdditionalImageStores {
-		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.imagestore=%s", config.Storage.Driver, s))
+	if len(config.Storage.Options.AdditionalImageStores) > 0 {
+		imageStore := config.Storage.Options.AdditionalImageStores[0]
+		additionalImageStores := strings.Join(config.Storage.Options.AdditionalImageStores, ",")
+		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.imagestore=%s", config.Storage.Driver, imageStore))
+		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.additionalimagestores=%s", config.Storage.Driver, additionalImageStores))
```

Comment:
I can’t see any parser of a driver `additionalimagestores` option; AFAICS that just doesn’t work.

---

All entries in `Options.AdditionalImageStores` have the same semantics (and that’s different from `Options.ImageStore`). I don’t know why we would treat the first item differently.

I _especially_ don’t know why we would now add the first item _twice_.

Also, if you look at the drivers’ option parsers, for the `overlay` driver both option names are treated the same, but for VFS, `additionalimagestore` is not recognized at all. So this breaks VFS.